### DISCUSSION TITLE
mail forwarding http routing endpoint issue

### DISCUSSION THREAD

**Discussion by andrewudxgen** [postalserver/postal#2524] [Help with using Postal]
hello,

we're routing all mails to http end point using routing module, it was working fine before.
now all the mail are forwarding to queue with status **PENDING**. we have attach the screens as reference.

when we try redeliver message button, it shows **"this message will be retried shortly"**

it works if we stop & start postal server from console.

thanks in advance!!!

![image](https://github.com/postalserver/docs/assets/40357654/39f3f5ab-d1a8-4351-8e14-edbb16ec5285)

![image](https://github.com/postalserver/docs/assets/40357654/991cc7c6-b1a6-4c6f-8699-7d6688edf416)

**Comment by andrewudxgen** [comment]
here is the postal status:

![image](https://github.com/postalserver/postal/assets/40357654/1be69d59-180a-47d9-a7bc-4c56701d2a68)

![image](https://github.com/postalserver/postal/assets/40357654/bdcbb388-93b1-495d-84db-a283c3486e37)

could you please help us the issue?

**Comment by rbustos** [comment]
Hi @andrewudxgen :

a) try following commands and check what logs shows:

`postal logs -f requerer `
`postal logs -f worker `

b) try restart process with:

`postal stop & postal start`

c) check into postal db:

`SELECT * FROM queued_messages`

if anything pending there then:

`UPDATE queued_messages SET locked_by = NULL, locked_at = NULL, retry_after = NULL;`

**Comment by andrewudxgen** [comment]
hello, thank you for quick response.

**postal logs -f requerer** ==> it doesnt show any reponse.
![image](https://github.com/postalserver/postal/assets/40357654/b7a92b13-7867-4014-8cae-9bce0268e8e8)

**postal logs -f worker**
![image](https://github.com/postalserver/postal/assets/40357654/3f875b4b-1f02-4988-a8ad-8be56a8c90a3)

**postal stop & postal start** ==> yes, when we do it, it works fine & after sometime, again mails are moving to with pending status and not routing to http end point.

thank you!!!

**Comment by rbustos** [comment]
@andrewudxgen clearly you have to dig into your MySQL server setup. Your log confirm that your Postal service lost communication with your database server.

Check your MySQL server logs to find out the culprit. Good luck :)

**Comment by willpower232** [comment]
As per your logs, it seems something is preventing the worker and requeuer containers from accessing the database, they require constant connections to both your database and rabbitmq so they terminate if connection is lost to either.

Presuming that they are both running somewhere else, you could try running `postal start` to restart them but if they return to a terminated state, it may be that your choice of server doesn't have enough memory to keep everything running.
